### PR TITLE
LeetCode-112 (Path Sum) solution added

### LINKED ISSUE #39: Add LeetCode - 112. Path Sum in C
I want to provide solution to the problem <a href="https://leetcode.com/problems/path-sum">Path Sum</a> in C

### PR DESCRIPTION
**Pull request by r8025n**
resolves #39

### CODE REVIEW

File: Algorithms/Easy/112_PathSum/Solution.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     struct TreeNode *left;
+ *     struct TreeNode *right;
+ * };
+ */
+
+bool hasPathSum(struct TreeNode* root, int targetSum){
```
Comment (Tahanima) [inline Algorithms/Easy/112_PathSum/Solution.c:10]: Change this code to this:
```c
bool hasPathSum(struct TreeNode* root, int targetSum) {
```

File: Algorithms/Easy/112_PathSum/Solution.c [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    if(root == NULL)
+        return false;
```
Comment (Tahanima) [inline Algorithms/Easy/112_PathSum/Solution.c:12]: Add a blank line after this, to separate 2 `if` statements

File: Algorithms/Easy/112_PathSum/Solution.c [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    if (root->left == NULL && root->right == NULL){
```
Comment (Tahanima) [inline Algorithms/Easy/112_PathSum/Solution.c:13]: Change this code to this:
```c
if (root->left == NULL && root->right == NULL) {
```

File: Algorithms/Easy/112_PathSum/Solution.c [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        if (root->val == targetSum)
+            return true;
```
Comment (Tahanima) [inline Algorithms/Easy/112_PathSum/Solution.c:15]: Add a blank line after `return true;`.

**Review by Tahanima — CHANGES_REQUESTED** [review]
Hi @r8025n 
Please resolve the requested changes and then, your pr will be ready for merge. Cheers! 🙌

File: Algorithms/Easy/112_PathSum/Solution.c [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        return false;
+    }
+    
+    return (hasPathSum(root->left, targetSum - root->val) || hasPathSum(root->right, targetSum - root->val));
+}
+
```
Comment (Tahanima) [inline Algorithms/Easy/112_PathSum/Solution.c:21]: Remove unnecessary blank lines.

**Comment by r8025n** [comment]
Hi
I have addressed the requested changes. Please review

**Review by Tahanima — APPROVED** [review]
Great work! :fire: Thank you so much for your contribution! :grinning: and welcome to **LeetCode Solution Curation** as a **Contributor**

Cheers! 🙌

**Comment by r8025n** [comment]
> Great work! fire Thank you so much for your contribution! grinning and welcome to **LeetCode Solution Curation** as a **Contributor**
> 
> Cheers! raised_hands

Thank you. Happy to contribute :)

**Pull request merged**